src/utils/request.ts

```diff
@@ -117,10 +129,56 @@ async function request(config: Config): Promise<OSDataHubResponse> {
     featureCount = output.results.length;
   }
 
-  logEndConditions(config);
+  logEndConditions(config, featureCount);
+
   if (typeof output === "undefined") {
     throw Error("There is no output at the end of request");
-  } else {
-    return output;
   }
+
+  return output;
+}
+
+async function requestNGD(config: Config): Promise<FeatureCollection> {
+  let endpoint: string;
+  let featureCount = 0;
+  let output: FeatureCollection | undefined = {
+    type: "FeatureCollection",
+    features: [],
+  };
+
+  const getEndpoint = config.paging.enabled
+    ? getOffsetEndpointNGD
+    : () => config.url;
+
+  const getData = config.method == "get" ? get : post;
+
+  while (continuePaging(config)) {
+    endpoint = getEndpoint(config, featureCount);
+
+    let response: Response = await getData(endpoint, config.key, config.body);
+
+    checkStatusCode(response.status);
+
+    const responseJson: FeatureCollection = <FeatureCollection>(
+      await response.json()
+    );
+
+    output.features = output.features.concat(responseJson.features);
+
+    if (responseJson.features && responseJson.features.length == 100) {
+      config.paging.position += 100;
+    } else {
+      config.paging.isNextPage = false;
+    }
```

Comment:
Should 100 be hardcoded?

Reply:
I think it's using length = 100 to check if it's reached the end of a page or not, so should be fine to keep I think